File: 152-maximum-product-subarray/152-maximum-product-subarray.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int maxi=nums[0], mins = nums[0], prod = nums[0],temp;
        for(int i = 1; i < nums.size(); i++){
            if(nums[i] < 0){
                temp = maxi;
                maxi = mins;
                mins = temp;
            }
            maxi = max(nums[i],maxi*nums[i]);
            mins = min(nums[i],mins*nums[i]);
            prod = max(prod,maxi);
        }
        return prod;
    }
};

File: 152-maximum-product-subarray/152-maximum-product-subarray.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        maxi = mins = prod = nums[0]
        for i in range(1,len(nums)):
            if(nums[i]<0):
                maxi,mins = mins,maxi
            maxi = max(nums[i],maxi*nums[i])
            mins = min(nums[i],mins*nums[i])
            prod = max(prod,maxi)
        return prod